informer/collector.go

```diff
@@ -0,0 +1,54 @@
+package informer
+
+import (
+	"context"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+var (
+	description *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName("operatorkit", "informer", "deletion_timestamp"),
+		"DeletionTimestamp of watched objects.",
+		[]string{
+			"exported_name",
+			"exported_namespace",
+		},
+		nil,
+	)
+)
+
+// Describe is used to describe metrics which are exported to Prometheus.
+func (i *Informer) Describe(ch chan<- *prometheus.Desc) {
+	ch <- description
+}
+
+// Collect is called by the Prometheus registry when collecting metrics from
+// the informer.
+func (i *Informer) Collect(ch chan<- prometheus.Metric) {
+	eventChan := make(chan watch.Event)
+	ctx := context.Background()
+	go func() {
+		err := i.fillCache(ctx, eventChan)
```

Comment:
I use `fillCache` here to get a list of all objects we watch.

Reply:
I'm not into the custom informer magic, but can't this cause a dead lock? Is `fillCache` synchronous?

Reply:
It is synchronous (unless I am totally misunderstanding something). 
How can I check or make sure that this doesn't cause a deadlock?

Reply:
Looking at the code, `fillCache` just starts a watch. Uncertain if that means we'll get a list of all objects.

Reply:
@xh3b4sd is probably the human to ask here

Reply:
I would not reuse `fillCache` here. Its name expresses a certain semantic and the chance that it does something you do not want to do all the time is pretty high, even though not given at the moment, but things can change. The reuse of it makes the code here also way more complicated than it has to be. I would just have copied the basics from `fillCache` and do the dance here again in combination with the metrics magic. The problem I see here already is a race between your for loop on `eventChan` below and the end of `fillCache` closing `eventChan`, which can cause the for loop below to panic. It is also that when a context is injected it should be used to cancel so we should cancel always if we are able to do it.

Reply:
I changed this and took the logic from `fillCache`. Can you take another look @xh3b4sd ?